### DISCUSSION TITLE
The bed leveling is damaging my bed

### DISCUSSION THREAD

**Discussion by abrahamfa** [1coderookie/KobraGoNeoInsights#21] [Q&A] [answered]
I'm using my Kobra Neo with Klipper.

I'm having issues with the mesh. I never get good results.

To be honest, I don't know what information from my code to provide. I'm using KAMP for the mesh. I have the x and y offsets activated and the zstop, along with the z offset, well configured. The results I'm getting are as follows:

![WhatsApp Image 2023-11-19 at 12 41 58 (1)](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/edf335fd-2155-49af-8fce-cde9f3d2b5f7)
![WhatsApp Image 2023-11-19 at 12 41 58](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/706215f2-268f-4351-8a3b-a35f3cd40907)
![WhatsApp Image 2023-11-19 at 12 41 57](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/e28dbb49-11fd-4a28-b532-6be0ec074670)
![Captura de pantalla 2023-11-19 204032](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/f4b771b3-8496-40d3-84c5-c2471d3b686b)

I don't know where else to look. My bed is clean; I've lubricated all the printer's wheels. I've adjusted the ABL to the height recommended by Anycubic, and I create the mesh with a low tolerance. I'm not sure if I'm missing something.

**Comment by NuclearPhoenixx** [comment]
Are you sure the z-offset is set correctly? Also your bed looks like it's crazy warped honestly. My first guess would be that that's at least a big part of the problem. Are you still using the rigid stock spacers or something else?

No idea about Klipper and KAMP yet, so maybe there is some problem there too. I'm sure @1coderookie can answer that.

**Reply by abrahamfa** [reply]
I would say that the Z offset, I think, should not be a problem. The lines don't adhere well in some places, and in other areas, they adhere too much. This seems to be a mesh issue at first glance. KAMP only does a bed to a custom range and saves time.

I also don't think it's bed warping. There are differences of 0.2 mm in the image I posted. I've had differences of over 2 mm before and managed to solve it. I don't have rubber spacers currently.

When the extruder passes through the areas where it doesn't adhere, I see the Z-axis move, and I suspect it's a configuration issue with the bed or something related to the X and Y offsets.

All of this is my understanding, but I might be mistaken about something.

**Comment by abrahamfa** [comment]
I just uninstalled Klipper and reverted to the original firmware. I thought of simplifying the configuration issues and checking if there could be something wrong with my printer during disassembly and reassembly. Indeed, there seems to be an issue with the hardware of my printer.

Right now, I won't know what it could be. I imagine that the X-axis might be incorrectly assembled. When I reassembled it, the belts were already somewhat tight, and I assume there's a problem there.

I'll thoroughly check the info site, but if you can shed some light on it, I would greatly appreciate it.

**Comment by 1coderookie** [comment]
Looking at the second picture with the skirt around the plate, the right side of the bed is too low / the nozzle too high. This can be caused if the x-gantry isn't perfectly trammed, the Y-brackets are crooked so that only one of the outer wheels properly touch the rail and then the inner wheel has been positioned too tight (trying to compensate the misaligned Y-bracket). Then the gantry moves up and when it moves down it lags and can stay too high. So that would be the very first thing.  Therefore give this a good reading and follow the steps in that violet expandable textbox there: https://1coderookie.github.io/KobraGoNeoInsights/hardware/axes/#tramming-the-x-axis-gantry    
Take your time and be as precise as you can, this one really is crucial!  

Do you have the rigid stock spacers or did you put in springs or silicone spacers?  

Next would be to use Klipper regularly, without KAMPS, and do a mesh (don't forget to save it!).  Make sure you load the mesh before an actual print job.   
I personally don't really see the huge advantage of KAMPS anyway, cuz *if you don't take off the PEI plate after a print* (just wait and let everything cool down, then the model pops right off), the whole thing doesn't really change. At least I personally never had issues with that.  
Once you've done that, print the above shown skirt again or print this one with a skirt around it: https://www.thingiverse.com/thing:5174552  
Then please post a pic of that print as well as a pic of your bedmesh again.   

Besides that, just as a note to your meshview shown above: take off the PEI plate and check that one. Is it slightly curved?  
If so, how was it placed on the bed - was it laying on your bed with the curve going upwards at the sides?

**Reply by abrahamfa** [reply]
The documentation is so extensive that I had missed this information. I love it, by the way. A tremendous effort.

For now, I'm going to start looking for some green bean cans to fine-tune this. This is my first printer, and it's second-hand. It comes mostly stock, and I'm tinkering with it.

I uninstalled Klipper for now to achieve a good first layer with the stock firmware. Yesterday, I downloaded the same STL you recommend to test this first layer.

Thank you so much! This afternoon, I'll try to adjust the printer, and I'll keep you informed.

**Reply by 1coderookie** [reply, marked as answer]
Yes, I know, it's quite a lot to read.. :(  
Thanks for your kind words :)  ..I'd appreciate if you'd starr the repo then if you like it btw ;)

You don't have to use *bean* cans, just make sure to use two objects of exactly the same height.  
Especially as it's second hand, go thru the steps being mentioned in the Calibration chapter. Start with a complete check of the frame itself - don't trust any preassembled parts! Most likely something is out of alignment, and then it's a PITA to find the reason for any weird behaviour. So take your time and check and tram *everything*. This will take a while and you have to take stuff apart, yes, but its really worth the hassle. 

As for the stock fw: I'd suggest to not use that one to check if everything is fine. That has it's own issues, like wrong settings about the probe's location and a Z-offset which will be forgotten somehow.  
So as you already used Klipper, go back using that one. Klipper is *much* more reliable.  

Btw: which printer.cfg did you use? Mine for the Neo from my repo?  

And btw2: are you the same guy who send me a dm via reddit about this issue? If so, using springs (you didn't answer my question above if you're using the rigid spacers or adjustable ones) doesn't solve your problems, it can actually make things worse and more complicated. So I'd also suggest to check the rigid stock spacers (if you have them) for being the same height as mentioned as one of the topics about calibrating the printer and then use those one for now. Then you can tram the x-gantry in relation to the bed - might be easier than tramming it in relation to the frame itslef (what you'd have to do when using adjustable spacers) and then tram the bed in relation to the head..

**Reply by NuclearPhoenixx** [reply]
> That has it's own issues, like wrong settings about the probe's location and a Z-offset which will be forgotten somehow.

I **never** once had this problem so far. Not sure what you did for this to happen. Only if I update the FW or do an ABL it resets the Z-Offset. But that's totally okay. For the probe offset he can just use one of the countless modded FWs.

Maybe it's easier to try with the stock FW at first, just so that you don't have to possibly debug multiple problems at the same time. If he's using stock or one of the mods, it'll print fine and he can be sure it's not the software causing weird issues. After that he can install Klipper again and double check everything with the configs there. At least that's how I'd do it and I think he had the same idea.

**Reply by 1coderookie** [reply]
Well, *I personally* didn't do anything to make it happen, it just happened. Many users have/had same issues..  

Sure, he can do whatever he feels to do is the best ;)

**Reply by abrahamfa** [reply]
I've used your config file named NEO_1coderookie_printer.cfg. I don't think I changed much except for the speeds of PROVE and BED_MESH, as with KAMP, it was quite slow to print and perform checks.

Regarding the spacers, they're the same as the factory ones. The printer only has one mod, which is the change of the fan vent. I noticed that one of them measured 0.1 mm more than the rest, and I corrected this.

Yesterday, I posted on Reddit, but I didn't reach out to you directly. I made a post in the Klipper community, thinking it was a configuration issue.

I believe I've sorted out my X-axis quite a bit. The advice has been really helpful. I've achieved good bed leveling except in some specific points:

![WhatsApp Image 2023-11-20 at 19 58 12](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/0ce47e48-2062-43ad-b356-d0a878706535)
![WhatsApp Image 2023-11-20 at 19 58 16](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/14b86564-1638-4ac7-a721-1a022e718ff7)

I still have areas where the offset is dropping a bit, but much less severe than the occasions I showed earlier. I'll continue experimenting these days, but I think I'm on the right track. I'll check if the Z-bar is straight and grease it in the coming days, as it's the only thing I haven't reviewed correctly.

While I'm on this post, I wanted to ask you a couple more questions:

1. In your guide, do you have instructions on how to level the extrusion screw? It confuses me a bit to have this screw and at the same time, have to find a good flow level.

2. From all the mods in your guide, which one would you recommend starting with, or do you consider essential to begin with?
![WhatsApp Image 2023-11-20 at 20 03 38](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/32fe007c-b984-4e3a-9d59-bc72881f94ee)


Thank you very much for everything!

**Reply by NuclearPhoenixx** [reply]
> In your guide, do you have instructions on how to level the extrusion screw? It confuses me a bit to have this screw and at the same time, have to find a good flow level.

Basically, you want the gear tension to be only as high as necessary as to not deform the filament, but on the other hand it should never be too loose as to slip. You can set it pretty easily by following these steps:
1. Loosen the screw so that tension is as low as possible
2. Now use the Prepare>Load Filament function from the menu (in Marlin, equivalent in Klipper)
3. When the extruder gear turns, put in your filament. It shouldn't move into the extruder, because tension is too low
4. Now gradually increase the tension until the gear catches the filament and is able to move it

It's from this guide: https://3dprinterly.com/extruder-tension-guide-for-3d-printing/

Personally, I adjusted the gear tension a little bit higher by slightly pulling on the filament while feeding and adjusting until it didn't slip anymore. That way it's a little bit more robust if you for some reason introduce more friction into the system, but it's still far away from deforming the filament itself. Don't forget to do all your flow and e-steps calibration after that.

**Reply by 1coderookie** [reply]
@abrahamfa Glad to hear it worked out for you.  
I cant judge the verall quality as you didn't provide a pic of the whole bedplate and the meshview, but I think you're on the right way, yes. If you have specific areas which  are 'much' lower, you can always try to put a piece of Kapton tape (or even a sticky PostIt-note) directly on the magnetic foil underneath the PEI plate in that location. Then probe again and see if it got better.  
It's also worth mentioning that it can help to heat up the bed and let it settle for like 5min or so vefore actually starting the print (and the mesh probing). I noticed clear differences when doing so.   

As fo the spring tension, I have some notes on that iirc, but I didn't describe a process I think. So you can proceed as described by @NuclearPhoenixx - I personally wouldn't loosen it all up tho, I'd just decrease the spring tension while extruding and see when it starts to slip. The (imho most) important is his personal note, as that's exactly how I'd do it: hold the filament kinda lightly between the tips of your thumb and index finger. You don't want to 'grab' it hard like a rope, you just want to add some resistance to the filament and feel how the gear pulls it in. Like pretending you're a spool which doesn't roll-off nice&smooth, just to be on the safe side. Then you should be good.  
What I use to do is to unload (I always do a cold pull) the filament then and check if I can notice some marks of the feeder gear's teeth. If you can barely see them being present and consistent, you're good. If you can see them really clear and the marks are already kinda deep in the filament, then go a bit lower with the tension. If you can't spot any marks at all, I personally would go a bit higher with the tension then.   
Note that the tension should be adjusted to the type of filament you're using. The softer the material is, the less tension should be used as well. This is extremely important when approaching TPU, as that can even tangle up in the gears if there's too much pressure. Besides that, as @NuclearPhoenixx already mentioned, if the pressure is too much, the filament will get deformed and artifacts will occur (which really can become an issue if one isn't aware of if - especially with dual-gear extruders; we only have one gear in here tho, but just fyi).  

As for the upgrades: hmm, where should I start and where should I end..? ;)  
1) I alsways insulate the bedplate right away, as it not only saves power (@NuclearPhoenixx measured it: ~25% less power consumption at 60° bedtemp!), but it keeps the temps of the bed much more stable. it also heats up faster and cools down slower. Especially when printing PETG at 80° bedtemp, imho this really has a positive effect, as PETG will come off the bedplate when the bedtemp drops.  
2) Print a better fan duct for the part cooling fan. The stock one really is bad. There are various models out there, I personally use one of those circular design. @NuclearPhoenixx posted his tests of different models here in the discussions (https://github.com/1coderookie/KobraGoNeoInsights/discussions/15), maybe read that. Pay attention to the position of the duct's outlet as well, as it seems that some are positioned a bit high.  
3) If you wanna invest some money, get yourself a plated copper heater block. Imho they do a better job than those aluminum ones, especially once you wanna print with a bit of higher speeds using a tuned Klipper.  
4) Get yourself a good nozzle. Means, don't go with those cheap 20pcs for 10bucks packages, just get one or two quality nozzles - I personally use original E3D nozzles. Depending on what you wanna print, you might wanna consider using a 0.6mm nozzle instead, as that will cut down your printing time. Going with a fine layer height like 0.2mm, you'll still be able to print fine detailed models and get a nice surface. If using abrasive filament, you should get a different nozzle than a pure brass one then, but that's all mentioned at the infosite as well..  
5) Maybe print that cover for the PSU which I made, you can find it at the site as well (chapter "Powersupply"). That one really makes the loud PSU fan much more quiet. --> **only do so on your own risk of course, see the comment on that below from @NuclearPhoenixx ;)** 

I think that's what I'd suggest in the first place for the printer itself.  

In addition to that, I personally would also recommend an enclosure. It helps keeping the fumes inside and it dampens the sound. It's also a nice protection agains dust. When printing PLA, you probably have to leave it open as PLA needs to be cooled down fast, but as soon as you start to use PETG, you'll notice the benefit: less smell, less noise and no risk of warping caused by cold airstreams hitting the machine.. 
I personally use these tent-style enclosures as I find them handy. See my notes about them here: https://1coderookie.github.io/KobraGoNeoInsights/hardware/other/#printer-enclosure  
I personally can HIGHLY recommend the tent from SainSmart as mentioned there, it's just sooo much better than those 'generic' tents.. 

Also keep an eye on your filament storage, you'll find some notes about that as well.  

And again: please consider starring this repo, it doesn't hurt or cost anything, but it does show a bit of appreciation. And if you wanna actively support my work, you can find a link to my Ko-fi account at the bottom of each page.. ;)  

Happy printing then!

**Reply by NuclearPhoenixx** [reply]
> Print that cover for the PSU which I made, you can find it at the site as well (chapter "Powersupply"). That one really makes the loud PSU fan much more quiet.

Oh yeah, about that -- I think it's pretty nice, works great and honestly probably nothing is going to happen if you swap it out. However, from a more safety-oriented POV I'd recommend against it. Yes it's much more quiet and easy to do if you watch out to not touch anything and unplug the PSU before of course. But keep in mind that the entire bottom side of the device is then uncovered in large parts (shocks), you just mounted a bunch of flammable material to it and removed a large part of the earthed surface which will also reduce the EMI shielding of this already very dubious quality power supply.

I'd love to change out the entire PSU for a quality brand for these reasons (this includes the noise), but I've not found one so far. Searching for this kind of 24V PSU mostly results in some generic brands (if at all) with broken english translations that all look alike unfortunately.

Like I said, it's most likely going to be fine and you technically shouldn't keep a 3D printer running unsupervised anyways, but just keep that in mind when doing something like this. You should also obviously know what you're doing if you're opening a PSU in general. You can also mount a different fan like it's mentioned in the docs, e.g. one with double ball bearings... that's a lot safer, but might not end up that much quieter.

**Reply by 1coderookie** [reply]
> But keep in mind that the entire bottom side of the device is then uncovered in large parts (shocks), you just mounted a bunch of flammable material to it and removed a large part of the earthed surface which will also reduce the EMI shielding of this already very dubious quality power supply.  

That's true! Only do on your own risk of course ;)  
To avoid the wide open surface, I added that mosquitto net to the inside, to prevent anything (bigger than the mosquitto net obviously) can get inside of it. But yeah, the flammable material and probably the EMI shielding is a clear CON argument, that's for sure.  

Changing the fan doesn't really help that much imho, as the stock fan itself really isn't that loud. It's the lid which is causing the noise, I guess also cuz they covered all of the ventilation holes. And as ppl do these kind of mods anyway, I thought I remix that original model to use it with the stock parts then..  

But yes, of course one should know what one is doing - and if not, better not approach it (or at least make sure to NOT touch any of the parts inside).  

You can look out for Meanwell PSUs tho, those use(d) to be quite good, but you probably pay nearly as much as you paid for the whole printer itself then (and maybe that will even come from the same factory - who knows).. :(

**Reply by NuclearPhoenixx** [reply]
> Changing the fan doesn't really help that much imho, as the stock fan itself really isn't that loud. It's the lid which is causing the noise, I guess also cuz they covered all of the ventilation holes. And as ppl do these kind of mods anyway, I thought I remix that original model to use it with the stock parts then..

Yeah sure, go for it. Unfortunate that it really doesn't help much to just switch the fan. I can't know, I haven't tested that yet.

> You can look out for Meanwell PSUs tho, those use(d) to be quite good, but you probably pay nearly as much as you paid for the whole printer itself then (and maybe that will even come from the same factory - who knows).. :(

I know it's not really economical, it's just something that would be nice to have. And you're probably right about them coming from the same factory. Just had a quick search and I found some by MW that also look identical from the outside. Having that mains voltage selection switch is a red flag to be honest. It just indicates lack of additional components and is more common either on pretty old designs or just extremely cheap devices that are designed to be sold globally for the minimum effort (ding ding ding). It's not like they all go up in flames after some time, but it's not really confidence inspiring either. At least there is probably some more QA and testing with MW... 🤷🏻

**Reply by abrahamfa** [reply]
I can now say that I have the bed leveled quite well. I'll share the last print of today after several tests.

![WhatsApp Image 2023-11-21 at 18 55 24](https://github.com/1coderookie/KobraGoNeoInsights/assets/76853642/00b5b8ea-88f3-466e-b6aa-bd307a22762e)


I've also adjusted the nut, and it looks pretty good, to be honest. It was a bit tricky, but with your advice, it wasn't too complicated! Of course, I have a few stringing issues now, so I imagine I need to move on to the next steps to adjust flow and retractions.

Regarding the extruder ventilation mod, currently, it has one similar to the one you've posted, although mine is completely circular. I think the one you're posting looks better, so once I have the printer well calibrated, I'll print this mod. I'll carefully consider the other mods you've recommended. The nozzle upgrade is probably something I'll go for.

I understand the flow test, but not so much about retractions. I'm using Orca for slicing, and I've seen that it has variable retraction towers. For speed, do I need to create a single tower with a retraction speed, right?

Thanks a lot!!!!!!!!

**Reply by 1coderookie** [reply]
Looks good! At least a lot better ;)  
I can't see it *that* well as it's a bit blurry when I zoom in, but it seems like you should lower your nozzle just a tiiiiny bit more (z-offset). I think I can spot individual lines here and there and also those incidents where the filament didn't stick were most likely caused by the nozzle being a bit too far away I think.  

If you already have the circular fan duct on there, then you're probably already good. Just give it a try then and see how it performs. The position really affects the outcome as well, so if it's a bit too high (which *might* usually be the case, not sure) then it doesn't cool the part properly. If it's too low tho, then you'll get problems like the duct may hit a part of the print. So just keep an eye on that..  

As for the further calibrations, please read the belonging sections at the infosite and check out those pages about calibrations I mentioned and linked to in the beginning of that chapter. I can't explain the basics in a few sentences properly and detailed, so better do a proper reading.   
However, before you start *anything*, check the notes about the retraction settings for your Neo! You don't want to use the 6mm retraction distance the profile from AC comes with for the Neo!!  
Then, as mentioned, find the best temperature for the filament first by printing temperature towers. Don't pay too much attention the the stringing there yet, as you'll do retraction towers then anyway.  
Once you found the correct temp, do a calibration of the e-steps first.  
Then go for the retraction tests and further calibrations.  
At least that's how I do it. But as I said, please read around for this by your own..

**Reply by abrahamfa** [reply]
Well, not much more to add. Time to calibrate!

I'll just open one last discussion in this thread while I'm here with you.

I've searched for information, but I can't find it, or I might not be searching well. How can I print in multiple colors with this type of printer? I've seen options for manually introducing filaments. Is there an automatic way? I assume it involves adding hardware, of course.

**Reply by NuclearPhoenixx** [reply]
Short answer is, you can't change filament/colors.

Long answer is you can manually change filaments by pausing the print and then feed from a new spool. I don't think there is any off-the-shelf hardware that can be retrofitted (especially not from Anycubic), so you'd have to play around on your own or follow some quite difficult DIY projects. You could theoretically also splice together multiple different lengths of filament and spool that for yourself to create your desired filament, but that would only work for the same type of filament, so only different colors. My best guess is that this is not practical at all. Probably pausing and switching out spools is your best bet. Just my two cents. Looking forward if @1coderookie does know any other ways.

https://all3dp.com/2/multi-material-3d-printing-an-overview/

**Reply by 1coderookie** [reply]
As @NuclearPhoenixx already mentioned, you could do so by pausing your print at the specific layer and then swap the filament manually. I personally never did it as I mostly do functional prints and don't need multi colour, but I've seen really good results of ppl doing it. If you're brave and more experienced, you could even change the type/material of filament by doing so.  

As an automatic solution, there's one project which might be worth being mentioned: it's ["The Enraged Rabbit Carrot Feeder (ERCF)"](https://github.com/EtteGit/EnragedRabbitProject). I didn't come across anybody who set this up for the Neo, but generally speaking it should be possible..
